lib/ssl/doc/src/using_ssl.xml

```diff
@@ -119,74 +124,77 @@ ok</code>
+  protocols such as SMTP, FTPS and HTTPS via a proxy.
+  </p>
+
+  <warning><p>Maximum security recommendations are however moving away from such solutions.</p></warning>
       
-      <p><em>Step 6:</em> Do the TLS handshake:</p>
-      <code type="erl">5 server> {ok, TLSSocket} = ssl:handshake(Socket, [{cacertfile, "cacerts.pem"},
-{certfile, "cert.pem"}, {keyfile, "key.pem"}]).
-{ok,{sslsocket,[...]}}</code>
+  <p>To upgrade to a TLS connection:</p>
+
+  <p><em>Step 1:</em> Start the server side:</p>
+  <code type="erl">1 server> ssl:start().
+  ok</code>
+
+  <p><em>Step 2:</em> Create a normal TCP listen socket and ensure
+  <c>active</c> is set to <c>false</c> and not set to any active mode
+  (except maybe once) before trying to upgrade a connection to a TLS
+  connection, otherwise TLS handshake messages can be delivered to the
+  wrong process.
+  </p>
+  <code type="erl">2 server> {ok, ListenSocket} = gen_tcp:listen(9999, [{reuseaddr, true}, {active, false}]).
+  {ok, #Port&lt;0.475&gt;}</code>
+
+  <p><em>Step 3:</em> Accept client connection:</p>
+  <code type="erl">3 server> {ok, Socket} = gen_tcp:accept(ListenSocket).
+  {ok, #Port&lt;0.476&gt;}</code>
+
+  <p><em>Step 4:</em> Start the client side:</p>
+  <code type="erl">1 client> ssl:start().
+  ok</code>
+
+  <code type="erl">2 client> {ok, Socket} = gen_tcp:connect("localhost", 9999,  [], infinity).</code>
+
+  <p><em>Step 5:</em> Do the TLS handshake:</p>
+  <code type="erl">4 server> {ok, TLSSocket} = ssl:handshake(Socket, [{cacertfile, "cacerts.pem"},
+  {certs_keys, [#{certfile => "cert.pem", keyfile => "key.pem"}
+  ]).
+  {ok,{sslsocket,[...]}}</code>
       
-      <p><em>Step 7:</em> Upgrade to a TLS connection. The client and
-      server must agree upon the upgrade. The server must call
-      <c>ssl:handshake/2</c> before the client calls
-      <c>ssl:connect/3.</c></p>
+  <p><em>Step 6:</em> Upgrade to a TLS connection. The client and
+  server must agree upon the upgrade. The server must be prepared to be a TLS server before the client
+  can do a successful connect.</p>
 
-      <code type="erl">3 client>{ok, TLSSocket} = ssl:connect(Socket, [{cacertfile, "cacerts.pem"},
-{certfile, "cert.pem"}, {keyfile, "key.pem"}], infinity).
+  <code type="erl">3 client>{ok, TLSSocket} = ssl:connect(Socket, [{cacertfile, "cacerts.pem"},
+  {certs_keys, [#{certfile => "cert.pem", keyfile => "key.pem"}], infinity).
 {ok,{sslsocket,[...]}}</code>
-      
-      <p><em>Step 8:</em> Send a message over TLS:</p>
+
+      <p><em>Step 7:</em> Send a message over TLS:</p>
       <code type="erl">4 client> ssl:send(TLSSocket, "foo").
-ok</code>
+      ok</code>
       
-      <p><em>Step 9:</em> Set <c>active true</c> on the TLS socket:</p>
-      <code type="erl">4 server> ssl:setopts(TLSSocket, [{active, true}]).
-ok</code>
+      <p><em>Step 8:</em> Set <c>active true</c> on the TLS socket:</p>
+      <code type="erl">5 server> ssl:setopts(TLSSocket, [{active, true}]).
```

Comment:
Isn't active=true something that is seldom used. We recommend active=once or active=N. So why do we have active=true in the example if we want to guide the users to the right track?

Reply:
This is still from the original example, it was used to make the example smaller. Focusing on the SSL/TLS part of the problem. But maybe we ought to change that too!